lcls_tools/common/measurements/tmit_loss.py

```diff
@@ -0,0 +1,258 @@
+
+        This method iterates through a dictionary of BPM objects and attempts
+        to fetch their TMIT buffer data using a specified buffer. If data
+        retrieval fails for a BPM, it is assigned a `None` value.
+
+        Args:
+            bpm_obj_dict (dict): A dictionary where keys are BPM element
+                                 names (str) and values are BPM objects.
+            my_buffer (BSABuffer): The buffer used to retrieve TMIT data for
+                                 each BPM.
+
+        Returns:
+            pd.DataFrame: A transposed DataFrame where:
+                          - Rows correspond to BPM elements.
+                          - Columns contain the retrieved TMIT buffer data.
+        """
+        data = {}
+
+        for element, bpm in self.bpms.items():
+            try:
+                # Get data from BSA buffer
+                bpm_data = bpm.tmit_buffer(self.my_buffer)
+                # If returned data is empty, skip
+                if len(bpm_data) == 0:
+                    pass
+                else:
+                    data[f"{element}"] = bpm_data
+            # If data cannot be retrieved, skip
+            # Retrieval errors happen when PVs are disconnected most commonly
+            except (BufferError, TypeError):
+                pass
+
+        df = pd.DataFrame(data)
+        return df.T
+
+    def get_bpm_idx(self, bpms_devices):
+        """
+        Retrieve the index positions of BPMs before and after the wire for a
+        given region.
+
+        This method selects predefined BPMs based on the specified region and
+        finds their corresponding indices in `bpms_devices`.
+
+        Args:
+            region (str): The region of interest. Must be one of:
+                          - "HTR", "DIAG0", "COL1", "EMIT2", "BYP",
+                            "SPD", "LTUS".
+            bpms_devices (list): A list of BPM device names.
+
+        Returns:
+            tuple: A tuple containing:
+                - list: Indices of BPMs located **before** the wire.
+                - list: Indices of BPMs located **after** the wire.
+        """
+        # Define valid regions
+        tmit_regions = {"HTR", "DIAG0", "COL1", "EMIT2", "BYP", "SPD", "LTUS"}
+        if self.region not in tmit_regions:
+            raise ValueError(
+                f"Invalid region '{self.region}'.Must be one of {{valid_regions}}"
+            )
+
+        bpms_before_wire = self.my_wire.metadata.bpms_before_wire
+        bpms_after_wire = self.my_wire.metadata.bpms_after_wire
+
+        # Create a lookup dictionary for index mapping
+        idx_map = {value: idx for idx, value in enumerate(bpms_devices)}
+
+        # Find indices of BPMs before and after the wire
+        idx_before = [idx_map[item] for item in bpms_before_wire if item in idx_map]
+        idx_after = [idx_map[item] for item in bpms_after_wire if item in idx_map]
+
+        return idx_before, idx_after
```

Comment:
I thought you added this info to the yaml file? could you not call them and index from the definition where needed?